packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -60,7 +60,10 @@ function resolveSync(
 
   if (REGEX_RELATIVE_IMPORT.test(target)) {
     // resolve relative import
-    const resolveTarget = path.resolve(basedir, target);
+    let resolveTarget = path.resolve(basedir, target);
+    if (target === '..' || target.endsWith('/')) {
+      resolveTarget += '/';
```

Comment:
this was lost in https://github.com/facebook/jest/pull/4325/files#diff-3887f79db6232183908265dcf84007acL60, introducing this bug

Reply:
not sure about the `target.endsWith('/')` but, what if the target is `.` instead of `./`?

Reply:
good call. I added a test for it, but I don't think it's the correct fix either

Reply:
How about checking for directory first? 
```diff
diff --git a/packages/jest-resolve/src/defaultResolver.ts b/packages/jest-resolve/src/defaultResolver.ts
index 40ded2c1f..131234b8c 100644
--- a/packages/jest-resolve/src/defaultResolver.ts
+++ b/packages/jest-resolve/src/defaultResolver.ts
@@ -60,10 +60,7 @@ function resolveSync(

   if (REGEX_RELATIVE_IMPORT.test(target)) {
     // resolve relative import
-    let resolveTarget = path.resolve(basedir, target);
-    if (target === '.' || target === '..' || target.endsWith('/')) {
-      resolveTarget += '/';
-    }
+    const resolveTarget = path.resolve(basedir, target);
     const result = tryResolve(resolveTarget);
     if (result) {
       return result;
@@ -100,7 +97,7 @@ function resolveSync(
     const dir = path.dirname(name);
     let result;
     if (isDirectory(dir)) {
-      result = resolveAsFile(name) || resolveAsDirectory(name);
+      result = resolveAsDirectory(name) || resolveAsFile(name);
     }
     if (result) {
       // Dereference symlinks to ensure we don't create a separate
```
Seems to work, but tests are still running locally.

Reply:
yeah, that was my original fix, see https://github.com/facebook/jest/pull/8912/commits/499e10a290461b3c8ddd90e3955a25f9397cdaed. But it's not that way upstream, which works correctly...

Reply:
Why does upstream not handle `'.'` though? :thinking:

Reply:
hah, good catch. That's a bug there, I opened up a PR. But that means I'm actually happy with this implementation 😅